### PR TITLE
Eth Click

### PR DESCRIPTION
**Pull request by agansari**
Enc28j60
 - corrects MAC registers
 - replaces NULL buffers with dummy buffers so some SPI drivers don't crash

LPCxpress55s69
 - replaces Flexcomm SPI's internal CS control with manual GPIO for Enc28j60

### CODE REVIEW

File: drivers/ethernet/eth_enc28j60.c [hunk 1]
```diff
@@ -382,18 +382,18 @@ static void eth_enc28j60_init_mac(struct device *dev)
 	}
```
Comment (jukkar) [inline drivers/ethernet/eth_enc28j60.c:382]: Please change the commit subject to `drivers: eth: enc28j60: Misc fixes`

File: drivers/ethernet/eth_enc28j60.c [hunk 2]
```diff
@@ -382,18 +382,18 @@ static void eth_enc28j60_init_mac(struct device *dev)
...
 
 	/* Configure MAC address */
-	eth_enc28j60_set_bank(dev, ENC28J60_REG_MAADR0);
-	eth_enc28j60_write_reg(dev, ENC28J60_REG_MAADR0,
+	eth_enc28j60_set_bank(dev, ENC28J60_REG_MAADR1);
```
Comment (jukkar) [inline drivers/ethernet/eth_enc28j60.c:385]: The enc28j60 was working mostly ok last time I tried without these changes. What are the symptoms if these changes are not done, could you describe these also in commit msg.

File: drivers/ethernet/eth_enc28j60.c [hunk 3]
```diff
@@ -510,6 +510,7 @@ static int eth_enc28j60_rx(struct device *dev, uint16_t *vlan_tag)
 	struct eth_enc28j60_runtime *context = dev->driver_data;
 	uint16_t lengthfr;
 	uint8_t counter;
+	uint8_t dummy[100];
```
Comment (jukkar) [inline drivers/ethernet/eth_enc28j60.c:513]: This is not good, we are using 100 bytes from stack here. Can't you just use the size of the buffer that is more suitable, it seems that we are reading max 4 bytes below, so why not have that.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: drivers/ethernet/eth_enc28j60.c (hunk 2 above)
Comment (agansari) [reply drivers/ethernet/eth_enc28j60.c:385]: It's a mismatch with the latest reference manual, there is no MAADR0 register.

File: drivers/ethernet/eth_enc28j60.c (hunk 3 above)
Comment (agansari) [reply drivers/ethernet/eth_enc28j60.c:513]: Sorry for that, spilled debug code, will change back to 4.

File: drivers/ethernet/eth_enc28j60.c (hunk 1 above)
Comment (agansari) [reply drivers/ethernet/eth_enc28j60.c:382]: Done

File: drivers/ethernet/eth_enc28j60.c (hunk 2 above)
Comment (jukkar) [reply drivers/ethernet/eth_enc28j60.c:385]: Ok. I just have not seen issues regarding Ethernet functionality, so wondering how this mismatch shows itself.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by MaureenHelm — APPROVED** [review]

**Review by galak — APPROVED** [review]

File: drivers/ethernet/eth_enc28j60.c [hunk 4]
```diff
@@ -600,13 +601,13 @@ static int eth_enc28j60_rx(struct device *dev, uint16_t *vlan_tag)
 		} while (frm_len > 0);
 
 		/* Let's pop the useless CRC */
-		eth_enc28j60_read_mem(dev, NULL, 4);
```
Comment (jfischer-phytec-iot) [inline drivers/ethernet/eth_enc28j60.c:603]: >Replaces NULL buffers as some SPI drivers will fail.

Why not fix the driver then? NULL is [valid](lhttps://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/spi.h#L171), @agansari please revert this changes and open a bug report regarding SPI drivers.

Comment (agansari) [reply drivers/ethernet/eth_enc28j60.c:603]: @jfischer-phytec-iot there are hal devices there are asserts that verify the return buffer is not NULL, this would mean moving the discard dummy buffer from one place (enc28j60) to another (spi drivers such as spi_mcux_flexcomm) and not all spi drivers were tested with NULL buffers.

Comment (jfischer-phytec-iot) [reply drivers/ethernet/eth_enc28j60.c:603]: > there are hal devices there are asserts that verify the return buffer is not NULL

That would not happen with a native driver.

> this would mean moving the discard dummy buffer from one place (enc28j60) to another (spi drivers such as spi_mcux_flexcomm)

sounds like "fixing a bug".  would you like to open a bug report for spi_mcux_flexcomm or provide a patch?

>and not all spi drivers were tested with NULL buffers.

But it is not enc28j60 (or any driver that uses SPI) fault. I cannot follow this approach. Why not fix the bug/issue but the symptoms?

Comment (jukkar) [reply drivers/ethernet/eth_enc28j60.c:603]: I agree with @jfischer-phytec-iot here, we should allow NULL here. If HAL is buggy, the fix should be added there. Sorry, I merged this PR too fast.
